docs/DeveloperGuide.md

```diff
@@ -1,34 +1,617 @@
+
+&nbsp;
+
+### Future Features
+
+1. Add weightage for modules.
+1. Integrate with github.
+1. Project info. (Including links and emails)
+1. Search via a filter.
+
+&nbsp;
+
+[🡅 Back to Table of Contents](#table-of-contents)
+
+----
+
+## Appendix: Requirements
+
+### Product Scope
+
+#### Target user profile:
+
+1. Needs a consolidated and personalisable workspace to organize their university modules
+1. Prefers desktop apps over other types
+1. Can type fast
+1. Is comfortable using CLI apps
+1. Is familiar with command-line shell environment
+
+#### Value proposition:
+
+Efficiently view and update regularly-needed information on modules and deadlines using a single keyboard.
+
+&nbsp;
+
+### User Stories
+
+> 💡 Priorities levels:<br>
+> `1`: High (Must have)<br>
+> `2`: Medium (Good to have)<br>
+> `3`: Low (Unlikely to have)
+
+| Priority | As a/an ... | I want to ... | So that I can ... |
+| --- | --- | --- | --- |
+| 1 | new user | see available commands | refer to the help page when I forget how to use the app |
+| 1 | NUS student | add a module | store useful information by module that I can easily refer to |
+| 1 | NUS student faced with e-learning | add a lesson | consolidate regularly-needed information such as Zoom links by tutorial/lecture, for quick access before the lesson |
+| 1 | busy NUS student | add a task | keep track of assignments and deadlines for a module in an organised to-do list |
+| 1 | NUS student | get an overview of the module / lesson / task list | filter out specific information with a single command |
+| 2 | NUS student | delete a module | store the information only temporarily, e.g. for the semester/term |
+| 2 | NUS SOC student | open a module’s cheat sheet(s) | I have a handy list of tools for the module, tests and exams at my disposal |
+| 2 | NUS SOC student with many team projects | View a module’s project team information and contact details | keep track of the various teams I am in and communicate more efficiently with my teammates |
+| 3 | busy NUS student | sort tasks by graded and done status | know which tasks are of highest priority |
+
+&nbsp;
+
+### Non-Functional Requirements
+
+1. Text editor will only work on OS with GUI support.
+1. All other features will work on any mainstream OS.
+1. It should work for students taking up to 10 modules.
+1. Each module should be able to store 100 tasks without issues.
+1. Every command should respond within 10s of input on a typical modern computer.
+
+&nbsp;
+
+### Glossary
+
+* Mainstream OS: Windows, Linux, Unix, OS-X
+* CLI: Command-Line Interface
+* GUI: Graphical User Interface
+* Module: A university module
+
+&nbsp;
+
+[🡅 Back to Table of Contents](#table-of-contents)
+
+----
+
+## Appendix: Instruction for Manual Testing
```

Comment:
Perhaps provide some instructions on how to reach edge / exception cases? Also add the exception messages when these cases are encountered